packages/pluggableWidgets/gallery-web/tests/e2e/specs/Gallery.spec.ts

```diff
@@ -0,0 +1,90 @@
+import page from "../../../../../../configs/e2e/src/pages/page";
+
+describe("gallery-web", () => {
+    beforeEach(() => {
+        page.open(); // resets page
+    });
+
+    describe("capabilities: sorting", () => {
+        it("applies the default sort order from the data source option", () => {
+            browser.setWindowRect(0, 0, 1200, 900);
+            const gallery = page.getWidget("gallery1");
+
+            gallery.waitForDisplayed({ timeout: 5000 });
+            browser.saveElement(gallery, "galleryContent");
+
+            expect(browser.checkElement(gallery, "galleryContent")).toEqual(0);
+        });
+
+        it("changes order of data choosing another option in the dropdown sort", () => {
+            const gallery = page.getWidget("gallery1");
+            const dropdownSort = page.waitForElement(".mx-name-gallery1 .dropdown-container .form-control", gallery);
+
+            dropdownSort.click();
+            const dropdownSortList = page.waitForElement(
+                ".mx-name-gallery1 .dropdown-container .dropdown-list",
+                gallery
+            );
+
+            dropdownSortList.$$("li")[4].click();
+            browser.saveElement(gallery, "galleryDropdownSort");
+
+            expect(browser.checkElement(gallery, "galleryDropdownSort")).toEqual(0);
+        });
+    });
+
+    describe("capabilities: filtering", () => {
+        it("filters by text", () => {
+            const gallery = page.getWidget("gallery1");
+            const textFilter = page.getElement(".mx-name-gallery1 .form-control", gallery);
+
+            textFilter.addValue("Leo");
+            browser.pause(1000);
+            browser.saveElement(gallery, "galleryTextFilter");
+
+            expect(browser.checkElement(gallery, "galleryTextFilter")).toEqual(0);
```

Comment:
why `toEqual(0)`? 🤔  What does that mean cuz it feels so weird 😂

Reply:
0 difference in the screenshot.